src/components/Experience.ts

```diff
@@ -29,11 +30,86 @@ export class Experience<TGameStartr extends FullScreenPokemon> extends GeneralCo
      *
      * @param pokemon   The pokemon to give experience to.
      * @param experience   The amount of experience to give.
+     * @returns True if the pokemon leveled up, false otherwise.
      */
-    public gainExperience(pokemon: IPokemon, experience: number): void {
+    public gainExperience(pokemon: IPokemon, experience: number): boolean {
         pokemon.experience += experience;
         if (pokemon.experience >= this.gameStarter.equations.experienceStarting(pokemon.title, pokemon.level + 1)) {
             this.levelup(pokemon);
+            return true;
         }
+        return false;
     }
+
+     /**
+      * Processes experience gain for one or more pokemon
+      *
+      * @param battleInfo   Information about the current battle.
+      * @param onComplete   Handler for when this is done.
+      *
+      */
+    public processBattleExperience(battleInfo: IBattleInfo, onComplete: () => void): void {
+        const experienceToGain =  this.gameStarter.equations.experienceGained(
+            battleInfo.teams.player, battleInfo.teams.opponent);
+
+        const experienceText: (string | string[])[][] = [[
+            battleInfo.teams.player.selectedActor.nickname,
+            ` gained ${experienceToGain} experience!`]];
+        this.gameStarter.menuGrapher.createMenu("GeneralText");
+        const levelUp = this.gainExperience(battleInfo.teams.player.selectedActor, experienceToGain);
+        if (levelUp) {
+            this.gameStarter.menuGrapher.addMenuDialog(
+                "GeneralText", experienceText,
+                () => this.processBattleLevelUp(battleInfo.teams.player.selectedActor, onComplete));
+        } else {
+            this.gameStarter.menuGrapher.addMenuDialog("GeneralText", experienceText, onComplete);
+        }
+        this.gameStarter.menuGrapher.setActiveMenu("GeneralText");
+    }
+
+     /**
+      * Processes level up for a given pokemon
+      *
+      * @param pokemon   Pokemon who is going to level up.
+      * @param onComplete   Handler for when this is done.
+      *
+      */
+    private processBattleLevelUp(pokemon: IPokemon, onComplete: () => void) {
+        this.gameStarter.menuGrapher.createMenu("GeneralText");
+        const experienceText: (string | string[])[][] = [[
+            pokemon.nickname,
+            ` grew to level ${pokemon.level}!`]];
+        if (this.canLearnMove(pokemon)) {
+            this.gameStarter.menuGrapher.addMenuDialog(
+                "GeneralText", experienceText,
+                () => this.learnBattleMove(pokemon, onComplete));
+        } else {
+            this.gameStarter.menuGrapher.addMenuDialog(
+                "GeneralText", experienceText, onComplete);
+        }
```

Comment:
As a general rule of thumb, I'd prefer it if you put an endline between `}`s and following lines of code. It's hard to read through the functions you've added because all the lines are together.